q2_moshpit/busco/busco.py

```diff
@@ -46,12 +49,21 @@ def _run_busco(
     """
     base_cmd = ["busco", *params]
 
-    manifest: pd.DataFrame = mags.manifest.view(pd.DataFrame)
-    manifest["sample_dir"] = manifest.filename.apply(
-        lambda x: os.path.dirname(x)
-    )
+    if isinstance(mags, MultiMAGSequencesDirFmt):
+        # Creates pandas df "manifest" from bins
+        manifest: pd.DataFrame = mags.manifest.view(pd.DataFrame)
+
+        # Make a new column in manifest with the directories of files
+        # listed in column "filename"
+        manifest["sample_dir"] = manifest.filename.apply(
+            lambda x: os.path.dirname(x)
+        )
+
+        # numpy.ndarray with unique dirs
+        sample_dirs = manifest["sample_dir"].unique()
```

Comment:
There _really_ is no need for all these comments - you wrote very good self-documenting code.
```suggestion
        manifest: pd.DataFrame = mags.manifest.view(pd.DataFrame)
        manifest["sample_dir"] = manifest.filename.apply(
            lambda x: os.path.dirname(x)
        )
        sample_dirs = manifest["sample_dir"].unique()
```